tests/conftest.py

```diff
@@ -78,77 +85,97 @@ def return_to_saved_path():
+        error_overlinking=boolify(error_overlinking_default),
+        error_overdepending=boolify(error_overdepending_default),
+        noarch_python_build_age=noarch_python_build_age_default,
+        enable_static=boolify(enable_static_default),
+        no_rewrite_stdout_env=boolify(no_rewrite_stdout_env_default),
+        ignore_verify_codes=ignore_verify_codes_default,
+        exit_on_verify_error=exit_on_verify_error_default,
+        conda_pkg_format=conda_pkg_format_default,
+    )
     assert result.no_rewrite_stdout_env is False
     assert result._src_cache_root is None
     assert result.src_cache_root == testing_workdir
     assert result.noarch_python_build_age == 0
     return result
 
 
-@pytest.fixture(scope='function')
+@pytest.fixture(scope="function")
 def testing_metadata(request, testing_config):
     d = defaultdict(dict)
-    d['package']['name'] = request.function.__name__
-    d['package']['version'] = '1.0'
-    d['build']['number'] = '1'
-    d['build']['entry_points'] = []
-    d['requirements']['build'] = []
-    d['requirements']['run'] = []
-    d['test']['commands'] = ['echo "A-OK"', 'exit 0']
-    d['about']['home'] = "sweet home"
-    d['about']['license'] = "contract in blood"
-    d['about']['summary'] = "a test package"
-    d['about']['tags'] = ['a', 'b']
-    d['about']['identifiers'] = 'a'
+    d["package"]["name"] = request.function.__name__
+    d["package"]["version"] = "1.0"
+    d["build"]["number"] = "1"
+    d["build"]["entry_points"] = []
+    d["requirements"]["build"] = []
+    d["requirements"]["run"] = []
+    d["test"]["commands"] = ['echo "A-OK"', "exit 0"]
+    d["about"]["home"] = "sweet home"
+    d["about"]["license"] = "contract in blood"
+    d["about"]["summary"] = "a test package"
+    d["about"]["tags"] = ["a", "b"]
+    d["about"]["identifiers"] = "a"
     testing_config.variant = get_default_variant(testing_config)
     testing_config.variants = [testing_config.variant]
     return MetaData.fromdict(d, config=testing_config)
 
 
-@pytest.fixture(scope='function')
+@pytest.fixture(scope="function")
 def testing_env(testing_workdir, request, monkeypatch):
-    env_path = os.path.join(testing_workdir, 'env')
-
-    check_call_env(['conda', 'create', '-yq', '-p', env_path,
-                    'python={}'.format(".".join(sys.version.split('.')[:2]))])
-    monkeypatch.setenv('PATH', prepend_bin_path(os.environ.copy(), env_path,
-                                                prepend_prefix=True)['PATH'])
+    env_path = os.path.join(testing_workdir, "env")
+
+    check_call_env(
+        [
+            "conda",
+            "create",
+            "-yq",
+            "-p",
+            env_path,
+            "python={}".format(".".join(sys.version.split(".")[:2])),
+        ]
+    )
+    monkeypatch.setenv(
+        "PATH",
+        prepend_bin_path(os.environ.copy(), env_path, prepend_prefix=True)["PATH"],
+    )
     # cleanup is done by just cleaning up the testing_workdir
```

Comment:
same thing what is the test actually testing?

Reply:
`conftest.py` is a special place for defining [pytest fixtures](https://docs.pytest.org/en/6.2.x/fixture.html), fixtures are pytest's way of implementing setup/teardown logic